pydra/engine/tests/test_workflow.py

```diff
@@ -0,0 +1,374 @@
+
+    wf.split(["x", "y"], x=[1, 2], y=[11, 12])
+    wf.combine("x")
+    wf.set_output([("out", wf.add2.lzout.out)])
+    wf.plugin = plugin
+
+    with Submitter(plugin=plugin) as sub:
+        sub.run(wf)
+
+    # checking the results
+    while not wf.done:
+        sleep(1)
+    results = wf.result()
+
+    assert results[0][0].output.out == 13
+    assert results[0][1].output.out == 24
+    assert results[1][0].output.out == 14
+    assert results[1][1].output.out == 26
+
+
+@pytest.mark.parametrize("plugin", Plugins)
+def test_wf_ndst_5(plugin):
+    """ workflow with two tasks, outer splitter and combiner on tasks level"""
+    wf = Workflow(name="wf_ndst_5", input_spec=["x", "y"])
+    wf.add(multiply(name="mult", x=wf.lzin.x, y=wf.lzin.y).split(["x", "y"]))
+    wf.add(add2(name="add2", x=wf.mult.lzout.out).combine("mult.x"))
+    wf.inputs.x = [1, 2]
+    wf.inputs.y = [11, 12]
+    wf.set_output([("out", wf.add2.lzout.out)])
+    wf.plugin = plugin
+
+    with Submitter(plugin=plugin) as sub:
+        sub.run(wf)
+
+    # checking the results
+    while not wf.done:
+        sleep(1)
+    results = wf.result()
+
+    assert results.output.out[0] == [13, 24]
+    assert results.output.out[1] == [14, 26]
+
+
+@pytest.mark.parametrize("plugin", Plugins)
+def test_wf_st_6(plugin):
+    """ workflow with three tasks, third one connected to two previous tasks,
+        splitter on the workflow level
+    """
+    wf = Workflow(name="wf_st_6", input_spec=["x", "y"])
+    wf.add(add2(name="add2x", x=wf.lzin.x))
+    wf.add(add2(name="add2y", x=wf.lzin.y))
+    wf.add(multiply(name="mult", x=wf.add2x.lzout.out, y=wf.add2y.lzout.out))
+    wf.split(["x", "y"], x=[1, 2, 3], y=[11, 12])
+
+    wf.set_output([("out", wf.mult.lzout.out)])
+    wf.plugin = plugin
+
+    with Submitter(plugin=plugin) as sub:
+        sub.run(wf)
+
+    # checking the results
+    while not wf.done:
+        sleep(1)
+    results = wf.result()
+
+    assert len(results) == 6
+    assert results[0].output.out == 39
+    assert results[1].output.out == 42
+    assert results[5].output.out == 70
+
+
+@pytest.mark.parametrize("plugin", Plugins)
+def test_wf_ndst_6(plugin):
+    """ workflow with three tasks, third one connected to two previous tasks,
+        splitter on the tasks levels
+    """
+    wf = Workflow(name="wf_ndst_6", input_spec=["x", "y"])
+    wf.add(add2(name="add2x", x=wf.lzin.x).split("x"))
+    wf.add(add2(name="add2y", x=wf.lzin.y).split("x"))
+    wf.add(multiply(name="mult", x=wf.add2x.lzout.out, y=wf.add2y.lzout.out))
+    wf.inputs.x = [1, 2, 3]
+    wf.inputs.y = [11, 12]
+
+    wf.set_output([("out", wf.mult.lzout.out)])
+    wf.plugin = plugin
+
+    with Submitter(plugin=plugin) as sub:
+        sub.run(wf)
+
+    # checking the results
+    while not wf.done:
+        sleep(1)
+    results = wf.result()
+
+    assert len(results.output.out) == 6
+    assert results.output.out == [39, 42, 52, 56, 65, 70]
```

Comment:
this looks like the workflow is auto-combining without a `combine`. is that intentional?

Reply:
no, they are not combined, just added additional tests with a combiner to show you the difference

Reply:
you can see all tests with names *_6/7/8/9, to see the difference in results with different combiners